src/utils.py

```diff
@@ -1,38 +1,45 @@
-import yaml
-import pickle
+"""
+    Utilities to load/save data.
+"""
+
 from typing import Dict, Any
+import pickle
+import yaml
+
 
 def load_training_params() -> Dict[str, Any]:
     """Loads and returns the training parameters.
 
     Raises:
-        FileNotFoundError: Raising FileNotFound if training_params.yaml is not found in the root folder of the project.
+        FileNotFoundError: Raising FileNotFound if training_params.yaml is
+        not found in the root folder of the project.
 
     Returns:
         Dict[str, Any]: Returns the training params as a dictionary.
     """
     try:
         with open('training_params.yaml', 'r') as file:
             params = yaml.safe_load(file)
-    except:
-        raise FileNotFoundError("Could not find file training_params.yaml.")
+    except BaseException as exception:
```

Comment:
Maybe it could be better to try and catch a more specific exception? I believe it is good practice to do so because this might catch an exception that is not related to the file not being found and this can become really confusing when trying to debug.